### DISCUSSION TITLE
Upgradeable Smart Contracts .   __UUPS_init()

### DISCUSSION THREAD

**Discussion by AllenGeorge08** [Cyfrin/foundry-full-course-cu#2051] [Q&A]
In the latest version of UUPSUpgradeable.sol there isn't a __UUPS_init() .....

```

contract BoxV1 is UUPSUpgradeable, Initializable, OwnableUpgradeable{
    uint256 internal number;
    

    //@custom: oz-upgrades-unsafe-allow constructor
    constructor(){
        _disableInitializers(); //Can't initialize variables in the constructor
    }

    function initialize() public initializer{
        __Ownable_init(msg.sender); //sets owner to msg.sender 
        __UUPSUpgradeable_init();
    }

    function getNumber() external view returns (uint256) {
        return number;
    }

    function version() external pure returns (uint256) {
        return 1;
    }

    function _authorizeUpgrade(address newImplementation) internal override{
    }
}
```

Need suggestions on whether i should skip that function. Also , Please explain what it exactly does.

**Comment by EngrPips** [comment]
Hello @AllenOps08, I believe the `__UUPS_init()` function is supposed to ensure the creation of a proper `UUPSUpgradeable` smart contract. The function `__UUPS_init()` inherits from `Initializable,` which ensures you don't mistakenly initialize your contract while deploying it. 

<img width="1440" alt="Screenshot 2024-07-18 at 11 30 55" src="https://github.com/user-attachments/assets/0f4a935b-dcad-4aff-92df-b9b920cd661c">

I opened the file on my end and looked up `__UUPS_init(),` and there was no comment on the function. There is also no implementation, but it only has an `onlyInitializing` modifier, which means you can't initialize your contract in its constructor. You can't initialize it more than once. You can ignore it, and I expect things to work as far as my knowledge encompasses.

**Reply by AllenGeorge08** [reply]
I fixed that, thanks bud.

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it!
